crates/bitwarden/src/crypto/enc_string.rs

```diff
@@ -11,6 +11,45 @@ use crate::{
 
 use super::{KeyDecryptable, KeyEncryptable, LocateKey};
 
+/// # Encrypted string primitive
+///
+/// [EncString] is a Bitwarden specific primitive that represents an encrypted string. They are
+/// are used together with the [KeyDecryptable] and [KeyEncryptable] traits to encrypt and decrypt
+/// data using [SymmetricCryptoKey]s.
+///
+/// The flexibility of the [EncString] type allows for different encryption algorithms to be used
+/// which is represented by the different variants of the enum.
+///
+/// ## Note
+///
+/// We are currently in the progress of splitting the [EncString] into distinct AES and RSA
+/// variants. To provide better control of which encryption algorithm is expected.
+///
+/// For backwards compatibility we will rarely if ever be able to remove support for decrypting old
+/// variants, but we should be opinionated in which variants are used for encrypting.
+///
+/// ## Variants
+/// - [AesCbc256_B64](EncString::AesCbc256_B64)
+/// - [AesCbc128_HmacSha256_B64](EncString::AesCbc128_HmacSha256_B64)
+/// - [AesCbc256_HmacSha256_B64](EncString::AesCbc256_HmacSha256_B64)
+/// - [Rsa2048_OaepSha256_B64](EncString::Rsa2048_OaepSha256_B64)
+/// - [Rsa2048_OaepSha1_B64](EncString::Rsa2048_OaepSha1_B64)
+///
+/// ## Serialization
+///
+/// [EncString] implements [Display] and [FromStr] to allow for easy serialization and uses a
+/// custom scheme to represent the different variants.
+///
+/// The scheme is one of the following schemes:
+/// - `[type].[iv]|[data]`
+/// - `[type].[iv]|[data]|[mac]`
+/// - `[type].[data]`
+///
+/// Where:
+/// - `[type]` is a single digit number representing the variant
```

Comment:
It's only a single digit number at the moment, if we ever add support for chacha20 or the like, we might need to expand to two digits, as we only have three values left. Maybe it's better to not specify the length?